pysph/tools/cull.py

```diff
@@ -0,0 +1,88 @@
+"""Culls files in a given directory, one in every 'c' files is spared.
+
+The specified directory can contain other directories that house the output
+ files; files in all these directories will be culled, sparing one in every
+ 'c' files. Note that DELETION IS PERMANENT.
+"""
+
+from pysph.tools.binder import find_sim_dirs, find_dir_size
+from pysph.solver.utils import get_files
+import os
+import sys
+import argparse
+
+
+def cull(src_path, c):
+
+    src_path = os.path.abspath(src_path)
+    sim_paths_list = find_sim_dirs(src_path)
+
+    initial_size = find_dir_size(src_path)
+
+    for path in sim_paths_list:
+        files = get_files(path)
+        safe_files = [files[i] for i in range(0, len(files), c)]
+        for f in files:
+            if f in safe_files:
```

Comment:
This is a bit inefficient and may not matter but you are doing a lookup in a list each time, so you could make `safe_files` a `set` to make this more efficient or change the logic to not require a lookup.

Reply:
Fixed, changed the array such that it stores the paths of the files to be deleted.